src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -569,10 +569,13 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                     _readOffset = 0;
                     _readLength = bytesRead;
                 }
+                else
+                {
+                    await FillAsync(async).ConfigureAwait(false);
```

Comment:
To make sure I understand, this was already done later as part of ReadNextREsponseHeaderLineAsync, but we're preemptively doing it here so that we can definitively say after this that data was received?

Reply:
Yes.